### PR TITLE
Sample async blocks for code-connection

### PR DESCRIPTION
**Pull request by joshlory**
### Sample program:

screen shot 2017-05-10 at 2 15 32 pm

screen shot 2017-05-10 at 2 18 03 pm

### Output:

screen shot 2017-05-10 at 2 15 39 pm

### CODE REVIEW

File: apps/src/codegen.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ exports.ForStatementMode = {
   UPDATE: 3
 };
 
+exports.asyncFunctionList = [];
```
Comment (islemaster) [inline apps/src/codegen.js:18]: Let's either add a descriptive comment describing how this exported array is to be used, or do something to make it obvious that it's private-ish / shouldn't be used directly.

File: apps/src/codegen.js [hunk 2]
```diff
@@ -266,6 +270,10 @@ export function marshalNativeToInterpreter(interpreter, nativeVar, nativeParentO
       nativeFunc: nativeVar,
       nativeParentObj: nativeParentObj,
     };
+    if (exports.asyncFunctionList.indexOf(nativeVar) !== -1) {
+      // Mark if this should be nativeIsAsync:
+      makeNativeOpts.nativeIsAsync = true;
+    }
```
Comment (islemaster) [inline apps/src/codegen.js:276]: I believe @pcardune is trying to gradually move this sort of behavior into `CustomMarshalingInterpreter` and `CustomMarshaler`, and gradually out of `codegen`. I don't see a super-clean way to do that yet here - maybe that's a good clean-up for a later PR.

**Comment by joshlory** [comment]
Happy to address these issues in a followup PR.  Any concerns with merging now to unblock Jingyu?

**Comment by islemaster** [comment]
:+1: :shipit: Sounds great.  Ship it.

**Pull request merged**